### DISCUSSION TITLE
Error in loading lora: "Invalid LoRA checkpoint"

### DISCUSSION THREAD

**Discussion by stjq12** [huggingface/diffusers#7901] [Q&A] [answered]
I try to load a lora model by official method "load_lora_weights", while raise the error "ValueError: Invalid LoRA checkpoint."

```
from diffusers import DiffusionPipeline
import torch
import sys

local = sys.path[0]

pipe = DiffusionPipeline.from_pretrained(local, torch_dtype=torch.float16, use_safetensors=True, variant="fp16")
pipe.load_lora_weights(".", weight_name = 'white_v10.safetensors')
pipe.to("cuda")

prompt = "xxx"

images = pipe(prompt=prompt).images[0]
```
err raise
`ValueError: Invalid LoRA checkpoint.`

Does anyone have a solution please?

**Comment by tolgacangoz** [comment]
Could you try with LoRA's absolute path?

**Reply by stjq12** [reply]
Yeah I have tried with the absolute path in `weight_name` but still the same. I think the pipeline have already get the right path.

**Reply by tolgacangoz** [reply]
Could you make absolute the left one, keep `weight_name='white_v10.safetensors'`?

**Reply by stjq12** [reply]
I try but it raise the same error..
`pipe.load_lora_weights(local + "\white_v10.safetensors", weight_name = local + 'white_v10.safetensors')`

**Comment by asomoza** [comment]
we won't be able to help if you don't provide the LoRA checkpoint or from where to download it.

**Reply by stjq12** [reply]
I download the LoRA from https://civitai.com/models/379435?modelVersionId=423637

**Reply by tolgacangoz** [reply]
Isn't it a full model, i.e., SDXL pipeline?

**Reply by stjq12** [reply]
Yes. sdxl-base-1.0 is the base model

**Reply by asomoza** [reply]
yep, that's not a LoRA, it's a full SDXL checkpoint.

**Reply by asomoza** [reply, marked as answer]
you don't load those as loras, you can load it with `pipe.from_single_file`

**Reply by stjq12** [reply]
Thanks bro!! I get it

**Reply by stjq12** [reply]
But I got another error. I use `pipe = StableDiffusionPipeline.from_single_file("white_v10.safetensors")` and the error raise with `StableDiffusionPipeline.__init__() got an unexpected keyword argument 'tokenizer_2'`. It this because the model is not suitable for the  StableDiffusionPipeline?

**Reply by tolgacangoz** [reply]
```py
pipe = StableDiffusionXLPipeline.from_single_file("white_v10.safetensors", torch_dtype=torch.float16).to("cuda")
```

**Reply by stjq12** [reply]
I Got it. I should use StableDiffusionXLPipeline.
